### PR TITLE
Add background music to EmulationStation (based off of RecalBox)

### PR DESCRIPTION
**Pull request by PokeEngineer**
I know it compiles without issues, but I can't test it out myself because I am bad at theming. I also know the "Enable Sounds" menu that I put in works.

### CODE REVIEW

**Comment by zigurana** [comment]
This is awesome! I was working on this as well, but with the GME library as
the music handler (chiptunes!).
Only, the earliest I can  test this will be in about a week or so, I'm
traveling atm.

Op di 28 feb. 2017 08:43 schreef Jacob Mullins <notifications@github.com>:

**Comment by Darknior** [comment]
Thanks a lot @PokeEngineer ... really cool idea :D 
They have some other cool options we can maybe take too ... have you plan to add some other ?

File: es-app/src/main.cpp [hunk 1]
```diff
@@ -68,7 +67,7 @@ bool parseArgs(int argc, char* argv[], unsigned int* width, unsigned int* height
 			Settings::getInstance()->setBool("Windowed", true);
 		}else if(strcmp(argv[i], "--vsync") == 0)
 		{
-			bool vsync = (strcmp(argv[i + 1], "on") == 0 || strcmp(argv[i + 1], "1") == 0) ? true : false;
+			bool vsync = (strcmp(argv[i + 1], "true") == 0 || strcmp(argv[i + 1], "1") == 0) ? true : false;
```
Comment (jrassa) [inline es-app/src/main.cpp]: Why change from on/off to true/false?  Not sure this change has anything to do with adding background music.

File: es-app/src/main.cpp [hunk 2]
```diff
@@ -102,7 +101,7 @@ bool parseArgs(int argc, char* argv[], unsigned int* width, unsigned int* height
 				"--debug				more logging, show console on Windows\n"
 				"--scrape			scrape using command line interface\n"
 				"--windowed			not fullscreen, should be used with --resolution\n"
-				"--vsync [1/on or 0/off]		turn vsync on or off (default is on)\n"
+				"--vsync [1/true or 0/false]		turn vsync on or off (default is true)\n"
```
Comment (jrassa) [inline es-app/src/main.cpp]: Same issue as above

File: es-app/src/main.cpp [hunk 3]
```diff
@@ -169,9 +168,6 @@ int main(int argc, char* argv[])
 	unsigned int width = 0;
 	unsigned int height = 0;
 
-	std::locale::global(boost::locale::generator().generate(""));
```
Comment (jrassa) [inline es-app/src/main.cpp]: Why remove this?

File: es-app/src/main.cpp [hunk 4]
```diff
@@ -18,7 +18,6 @@
 #include "Settings.h"
 #include "ScraperCmdLine.h"
 #include <sstream>
-#include <boost/locale.hpp>
```
Comment (jrassa) [inline es-app/src/main.cpp]: Why remove this?

**Comment by joolswills** [comment]
It looks as though this is more or less a backport of some patches from upstream and has some stuff mixed in. TBH it needs a bit of work before submitting it as a PR.

File: es-app/src/main.cpp (hunk 3 above)
Comment (PokeEngineer) [inline es-app/src/main.cpp]: Sorry... I based these changes off of RecalBox, and it was 3:00 in the morning when finishing these changes. I was tired and I will fix this.

**Pull request closed without merging**